### DISCUSSION TITLE
ERR: too many reordered frames

### DISCUSSION THREAD

**Discussion by cedricve** [bluenviron/gortsplib#471] [Questions]
Hey there,

I'm running one of the examples `client-read-format-h264-save-to-disk` using the `master` branch. I have the majority of my IP cameras working well, the RTSP streams read out properly, except one is throwing strange errors. Anyone an idea?

    2023/11/24 15:55:43 saved TS packet
    2023/11/24 15:55:43 saved TS packet
    2023/11/24 15:55:43 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 saved TS packet
    2023/11/24 15:55:44 140 RTP packets lost
    2023/11/24 15:55:44 122 RTP packets lost
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:44 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 ERR: too many reordered frames (20)
    2023/11/24 15:55:45 92 RTP packets lost
    2023/11/24 15:55:45 141 RTP packets lost
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:45 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 ERR: too many reordered frames (41)
    2023/11/24 15:55:46 51 RTP packets lost
    2023/11/24 15:55:46 82 RTP packets lost
    2023/11/24 15:55:46 66 RTP packets lost
    2023/11/24 15:55:46 4 RTP packets lost
    2023/11/24 15:55:46 ERR: too many reordered frames (62)
    2023/11/24 15:55:46 ERR: too many reordered frames (62)
    2023/11/24 15:55:46 ERR: too many reordered frames (62)
    2023/11/24 15:55:46 ERR: too many reordered frames (62)
    2023/11/24 15:55:46 ERR: too many reordered frames (62)
    2023/11/24 15:55:46 ERR: too many reordered frames (62)

**Comment by aler9** [comment]
Hello, the problem here is that there's not enough bandwidth between the camera and the server, and there are a lot of lost packets. Consequently, the DTS extraction algorithm get confused since it receives incomplete frames.

Increase the available bandwidth between the camera and the server, or lower the bitrate.

**Comment by cedricve** [comment]
Thank you, makes sense!
